### ISSUE TITLE
Add action to read the selected STT model

### ISSUE THREAD

**Issue by yannxaver** [mkiol/dsnote#189]
This is related to #186  

`flatpak run net.mkiol.SpeechNote --help-all`

```
--action <action>                    Invokes an <action>. Supported actions
                                       are: start-listening,
                                       start-listening-active-window,
                                       start-listening-clipboard,
                                       stop-listening, start-reading,
                                       start-reading-clipboard,
                                       pause-resume-reading, cancel,
                                       switch-to-next-stt-model,
                                       switch-to-prev-stt-model,
                                       switch-to-next-tts-model,
                                       switch-to-prev-tts-model, set-stt-model,
                                       set-tts-model.
```

I frequently need to change the STT model to dictate text in different languages. Changing the STT model is already possible using actions, which can be mapped to a hotkey. 

The problem is that I do not see a way to programmatically read the currently selected STT model. I either have to open the GUI or right click the tray icon. Both options are rather slow and inconvenient when using a keyboard-focused workflow.

If you added another action that returned the currently selected STT model it would be possible to build a custom status bar widget which always displays the selected STT model.

The same action might be useful for the TTS model. Personally, I am only using STT models at the moment.

**Comment by mkiol** [comment]
👍🏿 Will be in the next release.

**Comment by mkiol** [comment]
You will find the following options in the upcoming new version. If you need more or anything else, please let me know as I'm putting the final touches before the v4.7.0 release :)

```
$ ./dsnote --help
Usage: ./dsnote [options] [files...]

Options:
  --app-standalone          Runs in standalone mode (default). App and service
                            are not splitted.
  --app                     Starts app is splitted mode. App will need service
                            to function properly.
  --service                 Starts service only.
  --verbose                 Enables debug output.
  --print-state             Prints the current state.
  --print-available-models  Prints list of available STT and TTS models.
  --print-active-models     Prints the currently active STT and TTS models.
  --action <action>         Invokes an <action>. Supported actions are:
                            start-listening, start-listening-translate,
                            start-listening-active-window,
                            start-listening-translate-active-window,
                            start-listening-clipboard,
                            start-listening-translate-clipboard, stop-listening,
                            start-reading, start-reading-clipboard,
                            start-reading-text, pause-resume-reading, cancel,
                            switch-to-next-stt-model, switch-to-prev-stt-model,
                            switch-to-next-tts-model, switch-to-prev-tts-model,
                            set-stt-model, set-tts-model.
  --id <id>                 Language or model id. Used together with
                            start-listening-clipboard, start-reading-text,
                            set-stt-model or set-tts-model action.
  --text <text>             Text to read. Used together with
                            start-reading-text.
  --gen-checksums           Generates checksums for models without checksum.
                            Useful when adding new models to config file
                            manually.
  --hw-scan-off             Disables scanning for CUDA, ROCm, Vulkan, OpenVINO
                            and OpenCL compatible hardware. Use this option when
                            you observing problems in starting the app.
  --py-scan-off             Disables scanning for Python libraries. Use this
                            option when you observing problems in starting the
                            app.
  --reset-models            Reset the models configuration file to default
                            settings.
  --start-in-tray           Starts minimized to the system tray.
  --log-file <log-file>     Write logs to <log-file> instead of stderr.
  -h, --help                Displays help on commandline options.
  --help-all                Displays help including Qt specific options.
  -v, --version             Displays version information.

Arguments:
  files                     Text, Audio or Video files to open, optionally.
```

```
$ ./dsnote --print-active-models
Active STT model:
        sl_whisper_small_ft_samolego "Slovenščina (WhisperCpp Small Fine-Tuned Samolego) / sl"
Active TTS model:
        sl_piper_artur_medium        "Slovenščina (Piper Artur Medium Male) / sl"
```

```
$ ./dsnote --print-available-models
Available STT models: 4
        auto_fasterwhisper_base              "Auto detected (FasterWhisper Base) / auto"
        auto_fasterwhisper_large3_turbo      "Auto detected (FasterWhisper Large-v3 Turbo) / auto"
        sl_fasterwhisper_large2              "Slovenščina (FasterWhisper Large-v2) / sl"
        sl_whisper_small_ft_samolego         "Slovenščina (WhisperCpp Small Fine-Tuned Samolego) / sl"
Available TTS models: 6
        de_whisperspeech_small               "Deutsch (WhisperSpeech Small) / de"
        it_whisperspeech_small               "Italiano (WhisperSpeech Small) / it"
        lv_piper_aivars_medium               "Latviešu (Piper Aivars Medium Male) / lv"
        nl_whisperspeech_small               "Nederlands (WhisperSpeech Small) / nl"
        sl_piper_artur_medium                "Slovenščina (Piper Artur Medium Male) / sl"
        tr_piper_fettah_medium               "Türkçe (Piper Fettah Medium Male) / tr"
```

```
$ ./dsnote --print-state
State:
        3
Task state:
        0
```

Meaning of state numbers are explained here: https://github.com/mkiol/dsnote/blob/209ccbb4ef29516ab774030c49269b3e1aeb13b5/dbus/dsnote.xml.in#L19

**Comment by yannxaver** [comment]
Thank you for implementing these suggestions so quickly. 😊 It all looks great to me. 

My only contemplation was this: Each of the new commands returns two things at once. The commands about active and available models return STT and TTS information. And the command about state returns global state and task state.

If a consumer of these commands is only interested in one of the two things, it needs to parse the return value. This means any change in how the return value gets formatted by `dsnote` will be a breaking change to its consumers.

This is why I was contemplating if it could be beneficial to make the commands one level more granular.

```sh
--print-available-models <mode> Supported modes are stt, tts.  Prints list of available models
--print-active-model <mode> Supported modes are stt, tts. Prints the currently active model
--print-state <scope> Supported scopes are global, task. Prints the state of the provided application scope 
# maybe scope is a bad word for this. I'm not sure if I understand the distinction between the two states correctly
```
What are your thoughts on that?

**Comment by mkiol** [comment]
> What are your thoughts on that?

Sounds great :) Implemented in d75ea40e107a81c80ed773db903adfff360d6ffa. I'm "cooking" right now beta version, so if you want you will be able tomorrow test it by installing Speech Note from flathub-beta.

**Comment by mkiol** [comment]
The new version 4.7.0 is out and available on flathub. 

The new version includes new command line options:

- `--print-available-models`
- `--print-active-model`
- `--print-state`
